### PR TITLE
Reject packages with duplicate dependency groups

### PR DESCRIPTION
**Pull request by skofman1**
Addresses https://github.com/NuGet/NuGetGallery/issues/7714

### CODE REVIEW

File: src/NuGetGallery.Services/ServicesStrings.resx [hunk 1]
```diff
@@ -1130,4 +1130,7 @@ The {1} Team</value>
   <data name="RevokeCredential_UnrevocableApiKeyCredential" xml:space="preserve">
     <value>The API key credential with Key '{0}' is not revocable.</value>
   </data>
+  <data name="NuGetPackageDuplicateDependencyGroup" xml:space="preserve">
+    <value>A nuget package may not contain multiple dependency groups for the same target framework.</value>
```
Comment (loic-sharma) [inline src/NuGetGallery.Services/ServicesStrings.resx]: Suggestion:

```suggestion
    <value>A NuGet package may not contain multiple dependency groups with the same target framework.</value>
```

We could also say "The package may not..."

Comment (skofman1) [reply src/NuGetGallery.Services/ServicesStrings.resx]: I went for similar language as the other error message in that area. They all had 'nuget' and not 'NuGet'. Will use the rest of the suggestion :)

Comment (loic-sharma) [reply src/NuGetGallery.Services/ServicesStrings.resx]: That seems inconsistent with our branding. Would it be difficult to fix up all the other error messages?

Comment (skofman1) [reply src/NuGetGallery.Services/ServicesStrings.resx]: I'm sure there's plenty of lowercase nuget in the strings. However, it's out of scope of this PR..

**Comment by joelverhagen** [comment]
Have you checked if a duplicate dependency with a different version but in the same group has the same problem. I think it will and should be blocked similarly.

File: tests/NuGetGallery.Facts/Helpers/PackageHelperTests.cs [hunk 2]
```diff
@@ -167,6 +169,49 @@ public void ChecksIdVersionCombinedLength()
                 var ex = Assert.Throws<EntityException>(() => PackageHelper.ValidateNuGetPackageMetadata(metadata));
                 Assert.Contains("ID and version", ex.Message);
             }
+
+            [Fact]
+            public void ChecksDupliacteDependencyGroups()
+            {
+                var duplicateDependencyGroup = new PackageDependencyGroup(
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/Helpers/PackageHelperTests.cs]: Nit I would test this with a real nuspec/nupkg made in memory. We have some other tests that do this. Your test depends on groups of the same TFM not being clubbed together by the caller.

Comment (joelverhagen) [reply tests/NuGetGallery.Facts/Helpers/PackageHelperTests.cs]: If we have a test that verified this expectation I mentioned then this comment is less important. But afaik there's not a clear contract in this regard.

**Comment by skofman1** [comment]
> Have you checked if a duplicate dependency with a different version but in the same group has the same problem. I think it will and should be blocked similarly.

Tested the scenario 'a duplicate dependency with a different version but in the same group'. We get an error message:
image

If the dependency has the same version, they seem to be merged.

**Pull request merged**